framework/db/mssql/QueryBuilder.php

```
@@ -668,6 +668,24 @@ private function dropConstraintsForColumn($table, $column, $type = '')
+    /**
+     * {@inheritdoc}
+     *
+     * SQL Server does not support the `RECURSIVE` keyword for CTEs. Recursion is implicit when a CTE references itself.
+     */
+    public function buildWithQueries($withs, &$params)
+    {
+        if ($withs === null || $withs === []) {
+            return '';
+        }
+
+        foreach ($withs as $i => $_with) {
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$_with'. (undefined)

(UnusedLocalVariable)